plotly/figure_factory/_facet_grid.py

```diff
@@ -0,0 +1,1073 @@
+        col_values = df[facet_col].unique()
+        for row_count, x_val in enumerate(row_values):
+            for col_count, y_val in enumerate(col_values):
+                try:
+                    group = tuple_to_facet_group[(x_val, y_val)]
+                except KeyError:
+                    group = pd.DataFrame([[None, None]], columns=[x, y])
+                trace = graph_objs.Scatter(
+                    x=group[x].tolist(),
+                    y=group[y].tolist(),
+                    mode='markers',
+                    type=trace_type,
+                    marker=dict(
+                        color=DEFUALT_MARKER_COLOR,
+                        size=size,
+                        **kwargs_marker
+                    ),
+                    **kwargs
+                )
+                fig.append_trace(trace, row_count + 1, col_count + 1)
+                if row_count == 0:
+                    label = _return_label(col_values[col_count],
+                                          facet_col_labels,
+                                          facet_col)
+                    annotations.append(
+                        _annotation_dict(label, col_count + 1, num_of_cols,
+                                         row_col='col', flipped=flipped_cols)
+                        )
+
+            label = _return_label(row_values[row_count],
+                                  facet_row_labels,
+                                  facet_row)
+            annotations.append(
+                _annotation_dict(label, num_of_rows - row_count, num_of_rows,
+                                 row_col='row', flipped=flipped_rows)
+                )
+
+    # add annotations
+    fig['layout']['annotations'] = annotations
+
+    return fig
+
+
+def create_facet_grid(df, x, y, facet_row=None, facet_col=None,
+                      color_name=None, colormap=None, color_is_cat=False,
+                      widen_frame=False, facet_row_labels=None,
+                      facet_col_labels=None, height=700, width=700, size=5,
+                      trace_type='scatter', scales='fixed', dtick_x=None,
+                      dtick_y=None, show_boxes=True, **kwargs):
+    """
+    Returns figure for facet grid.
+
+    :param (pd.DataFrame) df: the dataframe of columns for the facet grid.
+    :param (str) x: the key of the dataframe to be used as the x axis df.
+    :param (str) y: the key of the dataframe to be used as the y axis df.
+    :param (str) facet_row: the key for row filter column for the facet grid.
+    :param (str) facet_col: the key for the column filter column for the facet
+        grid.
+    :param (str) color_name: the name of your dataframe column that will
+        function as the colormap variable.
+    :param (str|list|dict) colormap: the param that determines how the
+        color_name column colors the data. If the dataframe contains numeric
+        data, then a dictionary of colors will group the data categorically
```

Comment:
it's not "If the dataframe contains numeric data" it's more like "If `color_name` was set to a numerical column, then a continuous colorscale will be applied to the data. This colorscale can be customized by setting `colormap` to a Plotly colorscale name or through a custom colorscale. If `color_name` was set to a categorical column, then each unique value will be mapped a unique color. These colors can be customized by passing a dictionary that has unique values names as keys and color strings as variables. To learn more about colors and the type of a colorscale, run `help(plotly.colors)`."